pytorch_lightning/core/saving.py

```diff
@@ -361,9 +361,10 @@ def save_hparams_to_yaml(config_yaml, hparams: Union[dict, Namespace]) -> None:
     if not os.path.isdir(os.path.dirname(config_yaml)):
         raise RuntimeError(f'Missing folder: {os.path.dirname(config_yaml)}.')
 
-    if Container is not None and isinstance(hparams, Container):
+    if Container is not None and (isinstance(hparams, Container) or 
+                                  isinstance(v, Container) for v in hparams.values() ):
```

Comment:
Using OmegaConf.is_config() is cleaner in my opinion. it does the same through an official API.
```suggestion
    if Container is not None and (OmegaConf.is_config(hparams) or 
                                  OmegaConf.is_config(v) for v in hparams.values() ):
```

Reply:
Great thank you! Do you think its alright that we are calling OmegaConf.create() in terms of memory overhead?

Reply:
Yes, it should be fine for practical use cases.

Reply:
If you are changing to is_config(), make a pass on the rest oft he codebase to fix other instances too.

By the way:
Both the instance check for Container and the OmegaConf.is_config() are specific to OmegaConf 2.0 and newer (which means it requires Hydra 1.0 or newer for Hydra users).